suave/builder/builder.go

```diff
@@ -0,0 +1,77 @@
+package builder
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type builder struct {
+	config   *builderConfig
+	txns     []*types.Transaction
+	receipts []*types.Receipt
+	state    *state.StateDB
+	gasPool  *core.GasPool
+	gasUsed  *uint64
+}
+
+type builderConfig struct {
+	preState *state.StateDB
+	header   *types.Header
+	config   *params.ChainConfig
+	context  core.ChainContext
+}
+
+func newBuilder(config *builderConfig) *builder {
```

Comment:
@Ruteri curious your thoughts on if this is the best way to set up builder, I remember something non-obvious about reverts changing in builder recently so double check would be appreciated